src/main/java/lotto/domain/LotteryRankCounter.java

```diff
@@ -4,24 +4,24 @@
 import java.util.Objects;
 
 public class LotteryRankCounter {
-    private HashMap<LotteryRank, Integer> counter = new HashMap<>();
+    private final HashMap<LotteryRank, Long> counter;
+
+    LotteryRankCounter(HashMap<LotteryRank, Long> rankCounter) {
+        this.counter = rankCounter;
+    }
```

Comment:
특별한 구현체(`HashMap`)를 지정해서 이용하기 보다는, 가급적이면 포괄적인 상위 타입(`Map`)을 이용하면 좋습니다.
```suggestion
    private final Map<LotteryRank, Long> counter;

    LotteryRankCounter(Map<LotteryRank, Long> rankCounter) {
```
다행히도 지금은 stream collect가 `HashMap`을 반환했기에 (HashMap)캐스팅이 되었지만,
`Collections.synchronizedMap`이나 `Collections.unmodifiableMap`, 또는 `TreeMap`, `WeakHashMap` 같은 Map들은 이용할 수 없어요

Reply:
감사합니다. 미처 생각하지 못한 부분이었습니다!